src/collections/componentSet.ts

```diff
@@ -168,23 +171,62 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
   public static async fromManifest(input: string | FromManifestOptions): Promise<ComponentSet> {
     const manifestPath = typeof input === 'string' ? input : input.manifestPath;
     const options = (typeof input === 'object' ? input : {}) as Partial<FromManifestOptions>;
+    let manifestPre;
+    let manifestPost;
+    const hasManifestPre = !!options.destructivePre;
+    const hasManifestPost = !!options.destructivePost;
 
     const manifestResolver = new ManifestResolver(options.tree, options.registry);
     const manifest = await manifestResolver.resolve(manifestPath);
+    if (hasManifestPre) {
+      manifestPre = await manifestResolver.resolve(options.destructivePre);
+    }
+    if (hasManifestPost) {
+      manifestPost = await manifestResolver.resolve(options.destructivePost);
+    }
     const resolveIncludeSet = options.resolveSourcePaths
       ? new ComponentSet([], options.registry)
       : undefined;
     const result = new ComponentSet([], options.registry);
     result.apiVersion = manifest.apiVersion;
     result.fullName = manifest.fullName;
 
-    for (const component of manifest.components) {
+    const addComponent = (
+      component: MetadataComponent,
+      deletionType?: DestructiveChangesType
+    ): void => {
       if (resolveIncludeSet) {
-        resolveIncludeSet.add(component);
+        resolveIncludeSet.add(component, deletionType);
       }
       const memberIsWildcard = component.fullName === ComponentSet.WILDCARD;
       if (!memberIsWildcard || options.forceAddWildcards || !options.resolveSourcePaths) {
-        result.add(component);
+        result.add(component, deletionType);
+      }
+    };
+
+    for (const component of manifest.components) {
```

Comment:
write the manifest comps. and then the destructive post (if present) and the destructive pre (if present)